### PR TITLE
[DOCFIX] Add table of contents, update headers and fix unneeded line breaks 

### PR DESCRIPTION
**Pull request by waltonryan**
@yupeng9 @calvinjia This PR:

- Adds table of contents to each page
- drops headers down to at least `h2` where `h1` headers were used in body
- removes line breaks where not needed in the body text (partial fix).

Changes were made across `en`, `cn` and `pt`.

### CODE REVIEW

File: docs/cn/Building-Alluxio-Master-Branch.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ nickname: 构建主分支
 group: Resources
 ---
 
-* 内容列表
+* Table of Contents
```
Comment (yupeng9) [inline docs/cn/Building-Alluxio-Master-Branch.md]: Can you change this back to its Chinese translation?

Comment (yupeng9) [inline docs/cn/Building-Alluxio-Master-Branch.md]: Also, `内容列表` is the translation for `Table of Contents`, so the other changes in Chinese docs, could you also use the translation directly?

File: docs/cn/Running-Alluxio-Locally.md [hunk 2]
```diff
@@ -6,8 +6,6 @@ group: Deploying Alluxio
 priority: 1
 ---
 
-# 单机上独立运行Alluxio
```
Comment (yupeng9) [inline docs/cn/Running-Alluxio-Locally.md]: Why do we remove this?

File: docs/en/Configuring-Alluxio-with-GCS.md [hunk 3]
```diff
@@ -119,27 +122,27 @@ have the right access permission to the specified bucket, a permission denied er
 When Alluxio security is enabled, Alluxio loads the bucket ACL to Alluxio permission on the first
 time when the metadata is loaded to Alluxio namespace.
 
-### Mapping from GCS user to Alluxio file owner
+#### Mapping from GCS user to Alluxio file owner
```
Comment (yupeng9) [inline docs/en/Configuring-Alluxio-with-GCS.md]: Should this still be `###`?

File: docs/en/Configuring-Alluxio-with-GCS.md [hunk 4]
```diff
@@ -119,27 +122,27 @@ have the right access permission to the specified bucket, a permission denied er
...
 
 By default, Alluxio tries to extract the GCS user id from the credentials. Optionally,
 `alluxio.underfs.gcs.owner.id.to.username.mapping` can be used to specify a preset gcs owner id to
 Alluxio username static mapping in the format "id1=user1;id2=user2". The Google Cloud Storage IDs
 can be found at the console [address](https://console.cloud.google.com/storage/settings). Please use
 the "Owners" one.
 
-### Mapping from GCS ACL to Alluxio permission
+#### Mapping from GCS ACL to Alluxio permission
```
Comment (yupeng9) [inline docs/en/Configuring-Alluxio-with-GCS.md]: Should this still be `###`?

File: docs/en/Configuring-Alluxio-with-GCS.md [hunk 5]
```diff
@@ -119,27 +122,27 @@ have the right access permission to the specified bucket, a permission denied er
...
 
 Alluxio checks the GCS bucket READ/WRITE ACL to determine the owner's permission mode to a Alluxio
 file. For example, if the GCS user has read-only access to the underlying bucket, the mounted
 directory and files would have 0500 mode. If the GCS user has full access to the underlying bucket,
 the mounted directory and files would have 0700 mode.
 
-### Mount point sharing
+#### Mount point sharing
```
Comment (yupeng9) [inline docs/en/Configuring-Alluxio-with-GCS.md]: Should this still be `###`?

File: docs/en/Configuring-Alluxio-with-GCS.md [hunk 6]
```diff
@@ -119,27 +122,27 @@ have the right access permission to the specified bucket, a permission denied er
...
 
 If you want to share the GCS mount point with other users in Alluxio namespace, you can enable
 `alluxio.underfs.object.store.mount.shared.publicly`.
 
-### Permission change
+#### Permission change
```
Comment (yupeng9) [inline docs/en/Configuring-Alluxio-with-GCS.md]: Should this still be `###`?

File: docs/en/Key-Value-System-API.md [hunk 7]
```diff
@@ -40,21 +40,21 @@ Key-Value support in Alluxio is disabled by default, and it can be enabled in Al
...
 
 After enabling Key-Value support in Alluxio, you can run `./bin/alluxio runKVTest` to test whether
 the Key-Value system is running. You should see `Passed the test!` at the bottom of the output if
 the Key-Value system is correctly started.
 
-# Accessing Key-Value System in Java Application
+## Accessing Key-Value System in Java Application
 
-### Getting a Key-Value System Client
+#### Getting a Key-Value System Client
```
Comment (yupeng9) [inline docs/en/Key-Value-System-API.md]: Should this be `###`?

File: docs/cn/Building-Alluxio-Master-Branch.md (hunk 1 above)
Comment (waltonryan) [inline docs/cn/Building-Alluxio-Master-Branch.md]: will do.

File: docs/cn/Running-Alluxio-Locally.md (hunk 2 above)
Comment (waltonryan) [inline docs/cn/Running-Alluxio-Locally.md]: We don't need to place an `h1` in the docs pages as they come from the title of the doc in the `yml` front-matter. So in this case, it was not needed. Same for this header on the docs in `pt` and `en`

File: docs/en/Configuring-Alluxio-with-GCS.md (hunk 3 above)
Comment (waltonryan) [inline docs/en/Configuring-Alluxio-with-GCS.md]: yes.

File: docs/en/Configuring-Alluxio-with-GCS.md (hunk 4 above)
Comment (waltonryan) [inline docs/en/Configuring-Alluxio-with-GCS.md]: yes.

File: docs/en/Configuring-Alluxio-with-GCS.md (hunk 5 above)
Comment (waltonryan) [inline docs/en/Configuring-Alluxio-with-GCS.md]: yes.

File: docs/en/Configuring-Alluxio-with-GCS.md (hunk 6 above)
Comment (waltonryan) [inline docs/en/Configuring-Alluxio-with-GCS.md]: yes.

File: docs/en/Key-Value-System-API.md (hunk 7 above)
Comment (waltonryan) [inline docs/en/Key-Value-System-API.md]: yes.

**Comment by waltonryan** [comment]
@yupeng9 Thanks for the review. Addressed your feedback.

File: docs/cn/Running-Alluxio-Locally.md (hunk 2 above)
Comment (yupeng9) [inline docs/cn/Running-Alluxio-Locally.md]: Gotcha. Thanks.

**Pull request merged**